### PR TITLE
Updated monolog/monolog to 2.26.1

### PR DESCRIPTION
**Pull request by konarshankar07**
Depends on: https://github.com/magento/magento2-functional-testing-framework/pull/832

### Description (*)

Updated monolog/monolog to 2.26.1

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32868

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/555

### CODE REVIEW

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run Static Tests

File: composer.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
         "magento/composer": "1.6.0",
         "magento/magento-composer-installer": ">=0.1.11",
         "magento/zendframework1": "~1.14.2",
-        "monolog/monolog": "^1.17",
+        "monolog/monolog": "~1.26.1",
```
Comment (sivaschenko) [inline composer.json:68]: @konarshankar07 Thanks for the contribution! Can the library be updated by updating the composer.lock file only?

**Comment by sivaschenko** [comment]
Are there any partner contributors voluteers to collaborate on this PR and create corresponding EE PR to fix static tests?

File: composer.json (hunk 1 above)
Comment (sivaschenko) [inline composer.json]: Let's update monolog to the latest available version
```suggestion
        "monolog/monolog": "^2.2",
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @konarshankar07 ! Please see my comment

**Comment by sivaschenko** [comment]
Removed "help wanted" label, as @drpayyne has joined to help on the commerce repo side

**Comment by sivaschenko** [comment]
This pull request depends on the MFTF changes (update of the same library)

**Comment by sivaschenko** [comment]
EE pull request added to the related: https://github.com/magento/partners-magento2ee/pull/555

This pull request will be ready to merge once changes from https://github.com/magento/magento2-functional-testing-framework/pull/832 are released

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by jilu1** [comment]
@sivaschenko 
Is branch konarshankar07:update/package-monolog-1.26.1 ready with monolog upgrade? I got failed build http://10.234.231.143:8080/job/Functional-Tests-CE/4100/.

File: lib/internal/Magento/Framework/composer.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
         "laminas/laminas-uri": "^2.5.1",
         "laminas/laminas-validator": "^2.6.0",
         "magento/zendframework1": "~1.14.2",
-        "monolog/monolog": "^1.17",
+        "monolog/monolog": "~1.26.1",
```
Comment (sivaschenko) [inline lib/internal/Magento/Framework/composer.json]:
```suggestion
        "monolog/monolog": "^2.2",
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @konarshankar07 , can you please update the codebase for the latest monolog version.

Currently the tests are failing with error:

```
PHP Fatal error:  Declaration of Magento\Framework\Logger\Handler\Base::write(array $record) must be compatible with Monolog\Handler\StreamHandler::write(array $record): void in /var/www/html/lib/internal/Magento/Framework/Logger/Handler/Base.php on line 81
```

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by hostep** [comment]
Would love to see this happen, as monolog v2 comes with a [new datetime format that includes timezone information](https://github.com/Seldaek/monolog/commit/76a91c672213a17cb0de82f20e7a28c6f014eacf).

I was just researching this topic accidentally before running into this PR, because I think it could be very helpful in reading log files and I always get confused about the timestamps in Magento log files because it's currently not clear what timezone they are representing.

Alternatively, if this upgrade would take too much time to get approved, we could change the second parameter of the LineFormatter object that is created [over here](https://github.com/magento/magento2/blob/ab591597f860fdee5572d3894c649100e42d884c/lib/internal/Magento/Framework/Logger/Handler/Base.php#L54) to `Y-m-d\TH:i:sP`. That would also solve my need.

Just FYI 🙂

**Comment by andrewbess** [comment]
Hello @hostep 
We are actively working on updating the monolog to version 2.3.1 in new PRs.
This PR will be closed because of contributor doesn't continue working on this PR.
New PRs:
https://github.com/magento/magento2/pull/33497
https://github.com/magento/magento2-functional-testing-framework/pull/854

**Comment by sivaschenko** [comment]
Closing in favour to finalized https://github.com/magento/magento2/pull/33497

**Pull request closed without merging**